pkg/resources/nodeagent/daemonset.go

```diff
@@ -0,0 +1,242 @@
+// Copyright © 2021 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodeagent
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"strconv"
+
+	"emperror.dev/errors"
+	"github.com/banzaicloud/logging-operator/pkg/resources/templates"
+	"github.com/banzaicloud/logging-operator/pkg/sdk/api/v1beta1"
+	"github.com/banzaicloud/operator-tools/pkg/merge"
+	"github.com/banzaicloud/operator-tools/pkg/reconciler"
+	util "github.com/banzaicloud/operator-tools/pkg/utils"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	TailPositionVolume  = "positiondb"
+	BufferStorageVolume = "buffers"
+)
+
+func (n *nodeAgentInstance) daemonSet() (runtime.Object, reconciler.DesiredState, error) {
+	var containerPorts []corev1.ContainerPort
+	if n.nodeAgent.FluentbitSpec.Metrics != nil && n.nodeAgent.FluentbitSpec.Metrics.Port != 0 {
+		containerPorts = append(containerPorts, corev1.ContainerPort{
+			Name:          "monitor",
+			ContainerPort: n.nodeAgent.FluentbitSpec.Metrics.Port,
+			Protocol:      corev1.ProtocolTCP,
+		})
+	}
```

Comment:
Just a question: these ports are only required for the metrics server?

Reply:
Yes, and also for the liveness check

Reply:
Ok great. Thank you